tests/jupyterhub_api/test_create_session.py

```diff
@@ -0,0 +1,145 @@
+
+import json
+import requests
+import subprocess
+import os
+import argparse
+import sys
+sys.path.append("..")
+from SessionUtils import Test, JupyterhubTest
+from requests.packages.urllib3.exceptions import InsecureRequestWarning
+requests.packages.urllib3.disable_warnings(InsecureRequestWarning)
+import json
+import time
+
+
+def get_args():
+    parser = argparse.ArgumentParser(description='Arguments', formatter_class = argparse.ArgumentDefaultsHelpFormatter)
+    parser.add_argument("--port", dest="port", type=int,
+                        required = True,
+                        )
+
+    parser.add_argument( "--users", nargs='+', dest="users",
+                        required  = True,
+                        help='list of users')
+    parser.add_argument("--delay", dest="delay",
+                        required=True,
+                        help='delay while creating the server')
+
+    parser.add_argument("--json", dest="json",
+                         required=True,
+                         help = 'json data')
+
+    parser.add_argument("--base_path", dest="base",
+                        required=False,
+                        help='base path')
+    args = parser.parse_args()
+    return args
+
+
+class CreateSession(JupyterhubTest):
+    """
+    Create a session
+
+    To run the test run the following command:
+
+     python3 test_create_session.py --port 443 --users user3 --delay 30 --json '{"LCG-rel": "LCG_95a", "platform": "x86_64-centos7-gcc7-opt", "scriptenv": "none", "ncores": 2, "memory": 8589934592, "spark-cluster": "none"}'
+
+
+    For multiple users
+
+    python3 test_create_session.py --port 443 --users user0 user1 user2 --delay 30 --json '{"LCG-rel": "LCG_95a", "platform": "x86_64-centos7-gcc7-opt", "scriptenv": "none", "ncores": 2, "memory": 8589934592, "spark-cluster": "none"}'
+
+    """
+    def __init__(self, port, users, params, delay, base_path, verify):
+        JupyterhubTest.__init__(self, port, base_path, verify)
+        self.ref_test_name = "Session_creation_test"
+        super().log_params()
+        self.exit = 0
+        self.data = params
```

Comment:
What I previously said was that, IF you want to also log extra params, to avoid repeating the `log_params()`, just add these to a var `params` like so `self.params['users'] = users`, etc. And then, in `log_params()`, you do:

```
for param in self.params.keys:
    log("Param %s has value %s" % (param, self.params[param]) )
```